apps/opencs/view/render/pathgrid.cpp

```diff
@@ -0,0 +1,547 @@
+#include "pathgrid.hpp"
+
+#include <algorithm>
+
+#include <osg/Geode>
+#include <osg/Geometry>
+#include <osg/Group>
+#include <osg/PositionAttitudeTransform>
+#include <osg/Vec3>
+
+#include <components/sceneutil/pathgridutil.hpp>
+
+#include "../../model/world/cell.hpp"
+#include "../../model/world/commands.hpp"
+#include "../../model/world/commandmacro.hpp"
+#include "../../model/world/data.hpp"
+#include "../../model/world/idtree.hpp"
+
+namespace CSVRender
+{
+    class PathgridNodeCallback : public osg::NodeCallback
+    {
+        public:
+
+            virtual void operator()(osg::Node* node, osg::NodeVisitor* nv)
+            {
+                PathgridTag* tag = dynamic_cast<PathgridTag*>(node->getUserData());
```

Comment:
It makes no sense to dynamic_cast and then not checking the result of the cast.

One of the below would be fine:

<pre>
PathgridTag* tag = static_cast<PathgridTag*>(node->getUserData());
tag->getPathgrid()->update();
</pre>

<pre>
PathgridTag& tag = dynamic_cast<PathgridTag&>(*node->getUserData());
tag.getPathgrid()->update();
</pre>

<pre>
PathgridTag* tag = dynamic_cast<PathgridTag*>(node->getUserData());
if (tag)
   tag->getPathgrid()->update();
else
   std::cerr << "Warning, unable to run PathgridNodeCallback" << std::endl;
</pre>